### PR TITLE
Upgrade FontAwesome to 5.3.1

### PR DESCRIPTION
**Pull request by alandipert**
- This upgrades FontAwesome to a new major (by itself, breaking) version, but
  is backwards compatible because we include the [v4-shims](https://fontawesome.com/how-to-use/on-the-web/setup/upgrading-from-version-4) CSS that maps
  old names to new.
- `icon` tags are now browsable
- This is a step toward full V5 adoption that doesn't require us to
  come up with a plan for deprecating V4 icon names.

(Related: #2156 and #1966)

Confirmed that the FontAwesome CSS and webfont licenses didn't change between 4 and 5, added CC license for .svgs we now also include.

### CODE REVIEW

File: R/bootstrap.R [hunk 1]
```diff
@@ -1545,10 +1555,16 @@ icon <- function(name, class = NULL, lib = "font-awesome") {
   iconTag <- tags$i(class = iconClass)
 
   # font-awesome needs an additional dependency (glyphicon is in bootstrap)
-  if (lib == "font-awesome") {
-    htmlDependencies(iconTag) <- htmlDependency(
-      "font-awesome", "4.7.0", c(href="shared/font-awesome"),
-      stylesheet = "css/font-awesome.min.css"
+  if (lib %in% c("font-awesome", "fontawesome")) {
```
Comment (alandipert) [inline R/bootstrap.R]: Hm I'm not sure about this. Perhaps we should do this:

- If it's `font-awesome`, load the shim
- If it's `fontawesome`, don't load the shim

Then in subsequent versions of Shiny that ship V5 w/out the shim, specifying `font-awesome` is an error and `fontawesome` is the new deault `lib`

**Comment by eawachtel** [comment]
I'm really confused on what the current status of fontawesome in shiny is?  I'm using shiny v 1.0.5 and have download the fontawesome 4.7 and placed in www/shared.  I can not get most icons to work.  I see the svg files for the icons I want, but they do not work.  Any help appreciated

**Comment by jcheng5** [comment]
@eawachtel Please file a separate issue if you're having trouble with icons in 4.7.0. It'd be great if you could include a minimal reproducible example. Thanks!

**Comment by eawachtel** [comment]
I most certainly can provide an example of my code.

I use the below quite a bit just calling out different icon names in the
fontawesome library
actionButton("closelogin", label = NULL, icon = icon("times", lib=
"font-awesome"))

This works fine.

but if I reference a different icon "gas-pump"
actionButton("closelogin", label = NULL, icon = icon("gas-pump", lib=
"font-awesome"))

I get nothing

**Comment by alandipert** [comment]
@eawachtel I think `gas-pump` doesn't work because it was added in FontAwesome 5, and the currently-released version of Shiny only has FontAwesome 4.7. Once this PR is merged and we've released Shiny 1.2, `gas-pump` will work.

File: R/bootstrap.R (hunk 1 above)
Comment (alandipert) [reply R/bootstrap.R]: A half-baked scheme for supporting 4 and 5 in a subsequent release, but there's no reason to do it now. Removed and simplified.

File: R/bootstrap.R [hunk 2]
```diff
@@ -1545,10 +1555,16 @@ icon <- function(name, class = NULL, lib = "font-awesome") {
...
+    htmlDependencies(iconTag) <- list(
+      htmlDependency(
+        "fontawesome-all", "5.3.1", c(href="shared/fontawesome"),
+        stylesheet = "css/all.min.css"
+      ),
+      htmlDependency(
+        "fontawesome-v4-shims", "5.3.1", c(href="shared/fontawesome"),
+        stylesheet = "css/v4-shims.min.css"
+      )
```
[the remark this answers is not in the document]
Comment (alandipert) [reply R/bootstrap.R]: Makes sense. I did it 👍

**Comment by jcheng5** [comment]
@alandipert Are we missing "CC BY 4.0 License" in the LICENSE file? It looks like it applies to the SVG data, which we appear to be bundling.

**Pull request merged**